### DISCUSSION TITLE
How to stop Apache from caching API-Requests

### DISCUSSION THREAD

**Discussion by marudhupandiyang** [strapi/strapi#6425] [Legacy Topics]
**Describe the bug**
Recently i had created few new content types in local, took a db dump and restored the dump on server. Then started the server. My new content types were not appearing on the sidemenu under `Collection Types`, while it is shown under `plugin-> Content-type builder -> Collection Types`. Took a day to figure out it was due to the API request being cached.

**Steps to reproduce the behavior**

1. Have two strapi admin running. Assuming one is on production mode and the other is in development mode. 
2. Create new content types in the development mode strapi and dump the DB. Restore this dump on to the production version databasee. Also copy the new files from the development version to production version for strapi admin.
3. Now start the production version and see that under the sidebar `Collection Types`, new ones are not showing up.

**Expected behavior**
The new Content types should show up on the sidebar under Collection Types.

**Screenshots**
![Screenshot 2020-05-29 at 11 37 30 AM](https://user-images.githubusercontent.com/2351268/83227511-080c0200-a1a2-11ea-9bc9-1c938262309c.png)

**System**

- Node.js version: v12.16.2
- NPM version: 6.14.4
- Strapi version: v3.0.0-beta.20.1
- Database: MySQL Server 8.0.20-1debian10
- Operating system: Ubuntu

**Possible Solution**
The API responses should never be cached on the browser(client) or give a very small amount of time in future. The `expires` header in the response is currently set to +1 years and browser always retrieves the response from disk cache as show in the below and above image
![screenshot-devadmin justomerchantz com-2020 05 29-11_50_51](https://user-images.githubusercontent.com/2351268/83227984-01ca5580-a1a3-11ea-8dbd-7ca782f036b6.png)

**Comment by derrickmehaffy** [comment]
Just to clarify here, are you sitting behind a proxy like nginx? Because that certainly looks like some default caching coming from a proxy and not Strapi.

Edit: nevermind you are sitting behind an apache proxy: 
![image](https://user-images.githubusercontent.com/8593673/83229670-402d2980-a13d-11ea-8971-2ebca006fc29.png)

**Comment by derrickmehaffy** [comment]
Verifying this on a freshly installed, **not proxied**, local application:

![image](https://user-images.githubusercontent.com/8593673/83230045-f09b2d80-a13d-11ea-87e8-9133a8c62cce.png)

And Strapi does not apply caching. I'm going to move this over to a discussion as it's not a bug of Strapi but a question relating to eliminating the cache from the upstream proxy.

**Comment by marudhupandiyang** [comment]
That is right. I'm behind apache with expires module on. Not facing this on other applications, because the expires header is set on the response generated from the app server. Also as per apache docs [here ](https://httpd.apache.org/docs/2.4/mod/mod_expires.html)

> When the Expires header is already part of the response generated by the server, for example when generated by a CGI script or proxied from an origin server, this module does not change or add an Expires or Cache-Control header.

I hope it would be best to send up the expires/cache related headers from strapi to avoid these issues, since most people don't directly expose node and rather use web servers like `apache/ngnix` to proxy them.

**Reply by derrickmehaffy** [reply]
The thing is, Strapi has no caching system, thus it shouldn't be sending cache stuff directly.
